File: addons/crm_iap_mine/models/crm_lead.py
from odoo import _, fields, models


class CrmLead(models.Model):
    _inherit = "crm.lead"

    lead_mining_request_id = fields.Many2one(
        comodel_name="crm.iap.lead.mining.request",
        index="btree_not_null",
    )

    def _merge_get_fields(self):
        return super()._merge_get_fields() + ["lead_mining_request_id"]

    def action_generate_leads(self):
        return {
            "name": _("Need help reaching your target?"),
            "type": "ir.actions.act_window",
            "res_model": "crm.iap.lead.mining.request",
            "target": "new",
            "views": [[False, "form"]],
            "context": {"is_modal": True},
        }

File: addons/crm/static/tests/mock_server/mock_models/crm_lead.js
import { models } from "@web/../tests/web_test_helpers";

export class CrmLead extends models.ServerModel {
    _name = "crm.lead";
    _views = {
        search: /* xml */ `<search/>`,
        'form,false': /* xml */ `
            <form string="Lead">
                <sheet>
                    <field name="name"/>
                </sheet>
            </form>`
    };
}
